src/ast/passes/codegen_llvm.cpp

```diff
@@ -1404,29 +1404,31 @@ ScopedExpr CodegenLLVM::visit(Call &call)
   } else if (call.func == "strerror") {
     return visit(call.vargs.front());
   } else if (call.func == "strncmp") {
+    auto &left_arg = *call.vargs.at(0);
+    auto &right_arg = *call.vargs.at(1);
     auto size_opt = bpftrace_.get_int_literal(call.vargs.at(2));
```

Comment:
JC, is it possible and better to set the call expression size to be this min of the arg, left, and right in a earlier pass (semantic_analyser) ?

Reply:
The call to strncmp() returns an int64, so probably can't do it like that. But it is a good point that we should be trying to find a way to lower the AST into more unambiguous representations.